examples/model_selection/plot_roc.py

```diff
@@ -71,39 +59,69 @@
+from sklearn.metrics import roc_auc_score
+
+y_score_norm = y_score / y_score.sum(1, keepdims=True)
+unweighted_roc_auc_ovr = roc_auc_score(y_test, y_score_norm, multiclass="ovr")
+weighted_roc_auc_ovr = roc_auc_score(
+      y_test, y_score_norm, multiclass="ovr", average="weighted")
+print("One-vs-Rest ROC AUC scores: {0} (unweighted), {1} (weighted)".format(
+       unweighted_roc_auc_ovr, weighted_roc_auc_ovr))
+
+###############################################################################
+# Plotting the ROC curve for virginica
+# ....................................
+import matplotlib.pyplot as plt
+from sklearn.preprocessing import label_binarize
+from sklearn.metrics import roc_curve, auc
+
+# Binarize y_test to compute the ROC curve
+classes = np.unique(y)
+n_classes = len(classes)
+y_test_binarized = label_binarize(y_test, classes=classes)
+
 # Compute ROC curve and ROC area for each class
 fpr = dict()
 tpr = dict()
 roc_auc = dict()
 for i in range(n_classes):
-    fpr[i], tpr[i], _ = roc_curve(y_test[:, i], y_score[:, i])
+    fpr[i], tpr[i], _ = roc_curve(y_test_binarized[:, i], y_score[:, i])
     roc_auc[i] = auc(fpr[i], tpr[i])
 
-# Compute micro-average ROC curve and ROC area
-fpr["micro"], tpr["micro"], _ = roc_curve(y_test.ravel(), y_score.ravel())
+fpr["micro"], tpr["micro"], _ = roc_curve(
+      y_test_binarized.ravel(), y_score.ravel())
 roc_auc["micro"] = auc(fpr["micro"], tpr["micro"])
 
-
-##############################################################################
-# Plot of a ROC curve for a specific class
-plt.figure()
+fig, ax = plt.subplots()
 lw = 2
-plt.plot(fpr[2], tpr[2], color='darkorange',
-         lw=lw, label='ROC curve (area = %0.2f)' % roc_auc[2])
-plt.plot([0, 1], [0, 1], color='navy', lw=lw, linestyle='--')
-plt.xlim([0.0, 1.0])
-plt.ylim([0.0, 1.05])
-plt.xlabel('False Positive Rate')
-plt.ylabel('True Positive Rate')
-plt.title('Receiver operating characteristic example')
-plt.legend(loc="lower right")
-plt.show()
-
-
-##############################################################################
-# Plot ROC curves for the multiclass problem
-
-# Compute macro-average ROC curve and ROC area
+ax.plot(fpr[2], tpr[2], color='darkorange',
+        lw=lw, label='ROC curve (area = %0.2f)' % roc_auc[2])
+ax.plot([0, 1], [0, 1], color='navy', lw=lw, linestyle='--')
+ax.set_xlim([0.0, 1.0])
+ax.set_ylim([0.0, 1.05])
+ax.set_xlabel('False Positive Rate')
+ax.set_ylabel('True Positive Rate')
+ax.set_title('Receiver operating characteristic example for {}'.format(
+    iris.target_names[2]))
+ax.legend(loc="lower right")
+fig.show()
+
+###############################################################################
+# Plot ROC curves for the multiclass problem using One-vs-Rest
+# ............................................................
+# One can draw a ROC curve by considering each element of the label indicator
+# matrix as a binary prediction (micro-averaging).
```

Comment:
I know it's not you (it's the docstring description), but I personally don't understand what that means at all